src/github_permission_manager_demotion/handler.py

```diff
@@ -0,0 +1,127 @@
+        response = self.requests.patch(
+            f"https://api.github.com/repos/{repository}/issues/{issue_number}",
+            headers=self.auth_headers,
+            json={"state": "closed"}
+        )
+        print(f"Issue closed: {response.status_code}")
+
+    def post_comment_on_issue(self, repository, issue_number, comment_body):
+        data = {
+            "body": comment_body
+        }
+        response = requests.post(
+            f"https://api.github.com/repos/{repository}/issues/{issue_number}/comments",
+            headers=self.auth_headers,
+            json=data
+        )
+        print(f"Comment posted: {response.status_code}")
+
+    def get_all_org_owners(self, organization):
+        response = requests.get(
+            f"https://api.github.com/orgs/{organization}/members?role=admin",
+            headers=self.auth_headers
+        )
+        print(f"Request to get org owners: {response.status_code}")
+        org_owners = []
+        for owner in response.json():
+            print(f"Owner: {owner['login']}")
+            org_owners.append(owner['login'])
+        return org_owners
+
+
+    def make_member_on_github(self, organization, user):
+        data = {
+            "role": "member"
+        }
+        response = requests.put(
+            f"https://api.github.com/orgs/{organization}/memberships/{user}",
+            headers=self.auth_headers,
+            json=data
+        )
+        print(f"Comment posted: {response.status_code}")
+
+    def demote_user_lambda(self, event, _context):
+        self.initialise_aws_clients()
+        self.get_all_parameters()
+        organization = event.get('organization')
+        user = event.get('user')
+        repository = event.get('repository')
+        issue_number = event.get('issue_number')
+        org_owners = self.get_all_org_owners(organization)
+        if user in org_owners:
+            if is_last_org_owner(org_owners, user):
+                self.post_comment_on_issue(repository, issue_number, "User is the last owner - therefore will not be demoted")
+                self.close_issue(repository, issue_number)
+                return
+            self.post_comment_on_issue(repository, issue_number, "User is currently an owner - demotion to member in progress")
+            self.make_member_on_github(organization, user)
+            self.post_comment_on_issue(repository, issue_number, "User has been demoted")
+            self.close_issue(repository, issue_number)
+
+            table = self.dynamodb.Table('GithubElevationRequests')
+
+            # Update DynamoDB
+            table.update_item(
+                Key={'user': user},
+                UpdateExpression="set #status = :s, demoted_at = :t",
+                ExpressionAttributeNames={'#status': 'status'},
+                ExpressionAttributeValues={
+                    ':s': 'demoted',
+                    ':t': datetime.now().isoformat()
+                }
+            )
```

Comment:
can we flip this if statement as well please
`if user not in org_owners:` - log something and then return

Reply:
changed